### DISCUSSION TITLE
Container Image not working when built from self-hosted GitHub Runner

### DISCUSSION THREAD

**Discussion by nnellanspdl** [actions/virtual-environments#7617] [General questions]
Using a self-hosted runner, I am running a simple `docker build` command to build a Java sprint boot container.  The container image builds successfully.  However, when I try to run the container I get some strange errors about java beans missing.

If I take the same exact code, clone it locally, run the same exact `docker build` command, then that version of the image runs successfully without errors.

The only thing I can possibly think is that locally I am running Docker Engine 23 whereas on the GitHub Runner image it is still using Docker 20.

Has anyone else seen anything like this?

**Comment by vpolikarpov-akvelon** [comment]
Hi @nnellanspdl. What errors exactly do you get?

**Comment by nnellanspdl** [comment]
```
***************************
APPLICATION FAILED TO START
***************************
Description:
Parameter 0 of constructor in xxxController required a bean named 'yyyEntityManagerFactory' that could not be found.

Action:
Consider defining a bean named `yyyEntityManagerFactory' in your configuration.
```

**Reply by vpolikarpov-akvelon** [reply]
I'd assume that there is something with dependencies or code itself. But if it doesn't happen on your local machine I would recommend checking environment of the self-hosted runner once more. E.g. if it runs in isolated environment and/or uses libraries from mirror rather than public repo it may fetch some versions that are incompatible with your code. You may also compare build logs to see what are the differences in the build processes.
